Source/WebCore/css/CSSPrimitiveValue.cpp

```diff
@@ -1212,24 +1212,43 @@ String CSSPrimitiveValue::stringValue() const
     }
 }
 
+static NEVER_INLINE ASCIILiteral formatNonfiniteValuePrefix(double number)
+{
+    if (number == std::numeric_limits<double>::infinity())
+        return "infinity"_s;
+    if (number == -std::numeric_limits<double>::infinity())
+        return "-infinity"_s;
+    ASSERT(std::isnan(number));
+    return "NaN"_s;
+}
+
+static NEVER_INLINE void formatNonfiniteValue(StringBuilder& builder, double number, ASCIILiteral suffix)
+{
+    return builder.append(formatNonfiniteValuePrefix(number), suffix.isEmpty() ? ""_s : " * 1"_s, suffix);
+}
+
 static NEVER_INLINE String formatNonfiniteValue(double number, ASCIILiteral suffix)
 {
-    auto prefix = [&] {
-        if (number == std::numeric_limits<double>::infinity())
-            return "infinity"_s;
-        if (number == -std::numeric_limits<double>::infinity())
-            return "-infinity"_s;
-        ASSERT(std::isnan(number));
-        return "NaN"_s;
-    }();
-    return makeString(prefix, suffix.isEmpty() ? ""_s : " * 1"_s, suffix);
+    return makeString(formatNonfiniteValuePrefix(number), suffix.isEmpty() ? ""_s : " * 1"_s, suffix);
+}
+
+NEVER_INLINE void CSSPrimitiveValue::formatNumberValue(StringBuilder& builder, double value, ASCIILiteral suffix)
```

Comment:
Does this need to be a member function?

Reply:
Nope. And I am planning on moving all this to a new CSSUnit specific home soon.